devices/AtomQrCode/QrCodeReader.cs

```diff
@@ -0,0 +1,843 @@
+            {
+                throw new ArgumentException();
+            }
+
+            SendCommandType07(suffix);
+
+            // store new timeout value
+            _lastTimeout = timeout;
+        }
+
+        private void SendCommandType07(
+            byte[] suffix,
+            bool sendWakeUp = false)
+        {
+            // buffer content length
+            int commandLength = 1;
+
+            // set command prefix
+            _commandBuffer[0] = _commandPrefix07;
+
+            // set command common sequence
+            // this type has different ending, so removing the last one 
+            Array.Copy(
+                _commandCommon,
+                0,
+                _commandBuffer,
+                commandLength,
+                _commandCommon.Length - 1);
+
+            // update counter
+            commandLength += _commandCommon.Length - 1;
+
+            // set command suffix
+            Array.Copy(
+                suffix,
+                0,
+                _commandBuffer,
+                commandLength,
+                suffix.Length);
+
+            // update counter
+            commandLength += suffix.Length;
+
+            SendCommand(
+                _commandBuffer,
+                0,
+                commandLength,
+                sendWakeUp);
+        }
+
+        private void SendCommandType08(
+            byte[] suffix,
+            bool sendWakeUp = false)
+        {
+            // buffer content length
+            int commandLength = 1;
+
+            // set command prefix
+            _commandBuffer[0] = _commandPrefix08;
+
+            // set command common sequence 
+            Array.Copy(
+                _commandCommon,
+                0,
+                _commandBuffer,
+                commandLength,
+                _commandCommon.Length);
+
+            // update counter
+            commandLength += _commandCommon.Length;
+
+            // set command suffix
+            Array.Copy(
+                suffix,
+                0,
+                _commandBuffer,
+                commandLength,
+                suffix.Length);
+
+            // update counter
+            commandLength += suffix.Length;
+
+            SendCommand(
+                _commandBuffer,
+                0,
+                commandLength,
+                sendWakeUp);
+        }
+
+        private void SendCommand(
+            byte[] buffer,
+            int offset,
+            int count,
+            bool sendWakeUp = false)
+        {
+            if (!_readerSerialPort.IsOpen)
+            {
+                throw new InvalidOperationException();
+            }
+
+            if (sendWakeUp)
+            {
+                // send wake-up 
+                _commandBuffer[_commandBufferLastPos] = 0x00;
+
+                _readerSerialPort.Write(
+                    _commandBuffer,
+                    _commandBufferLastPos,
+                    1);
+
+                // need to wait
```

Comment:
just add that this data is from the datasheet

Reply:
nothing  from the datasheet here...